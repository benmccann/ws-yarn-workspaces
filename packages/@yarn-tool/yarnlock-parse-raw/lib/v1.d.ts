/// <reference types="node" />
import { stringify } from '@yarnpkg/lockfile';
import { IYarnLockRawSourceV1Full } from '@yarn-tool/yarnlock-types';
declare module '@yarnpkg/lockfile' {
    function parse<T extends IYarnLockRawSourceV1Full = IYarnLockRawSourceV1Full>(source: string, fileLoc?: string): T;
    function parse(source: string, fileLoc?: string): IYarnLockRawSourceV1Full;
}
export declare function parseYarnLockRawV1<T extends IYarnLockRawSourceV1Full = IYarnLockRawSourceV1Full>(source: string | Buffer, fileLoc?: string): T;
export declare function parseYarnLockRawV1Root(source: string | Buffer, fileLoc?: string): import("@yarn-tool/yarnlock-types").IYarnLockDataRecord<import("@yarn-tool/yarnlock-types").IYarnLockDataRowV1>;
export { stringify as stringifyYarnLockRawV1 };
