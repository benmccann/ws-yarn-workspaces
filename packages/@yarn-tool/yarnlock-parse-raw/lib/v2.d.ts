/// <reference types="node" />
import { stringifySyml } from '@yarnpkg/parsers';
import { IYarnLockRawSourceV2 } from '@yarn-tool/yarnlock-types';
declare module '@yarnpkg/parsers' {
    function parseSyml<T extends IYarnLockRawSourceV2 = IYarnLockRawSourceV2>(source: string): T;
    function parseSyml(source: string): IYarnLockRawSourceV2;
}
export declare function parseYarnLockRawV2<T extends IYarnLockRawSourceV2 = IYarnLockRawSourceV2>(source: string | Buffer): T;
export { parseYarnLockRawV2 as parseYarnLockRawV2Root };
export { stringifySyml as stringifyYarnLockRawV2 };
