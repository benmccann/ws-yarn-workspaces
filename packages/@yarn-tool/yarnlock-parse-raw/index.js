"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stringifyYarnLockRawV1 = exports.parseYarnLockRawV1 = exports.stringifyYarnLockRawV2 = exports.parseYarnLockRawV2 = void 0;
var v2_1 = require("./lib/v2");
Object.defineProperty(exports, "parseYarnLockRawV2", { enumerable: true, get: function () { return v2_1.parseYarnLockRawV2; } });
Object.defineProperty(exports, "stringifyYarnLockRawV2", { enumerable: true, get: function () { return v2_1.stringifyYarnLockRawV2; } });
var v1_1 = require("./lib/v1");
Object.defineProperty(exports, "parseYarnLockRawV1", { enumerable: true, get: function () { return v1_1.parseYarnLockRawV1; } });
Object.defineProperty(exports, "stringifyYarnLockRawV1", { enumerable: true, get: function () { return v1_1.stringifyYarnLockRawV1; } });
//# sourceMappingURL=index.js.map