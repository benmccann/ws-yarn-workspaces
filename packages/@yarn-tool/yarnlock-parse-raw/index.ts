export { parseYarnLockRawV2, stringifyYarnLockRawV2 } from './lib/v2';
export { parseYarnLockRawV1, stringifyYarnLockRawV1 } from './lib/v1';
