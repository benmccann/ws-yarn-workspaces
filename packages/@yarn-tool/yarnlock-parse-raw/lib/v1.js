"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stringifyYarnLockRawV1 = exports.parseYarnLockRawV1Root = exports.parseYarnLockRawV1 = void 0;
const lockfile_1 = require("@yarnpkg/lockfile");
Object.defineProperty(exports, "stringifyYarnLockRawV1", { enumerable: true, get: function () { return lockfile_1.stringify; } });
function parseYarnLockRawV1(source, fileLoc) {
    return (0, lockfile_1.parse)(source === null || source === void 0 ? void 0 : source.toString(), fileLoc);
}
exports.parseYarnLockRawV1 = parseYarnLockRawV1;
function parseYarnLockRawV1Root(source, fileLoc) {
    return parseYarnLockRawV1(source, fileLoc).object;
}
exports.parseYarnLockRawV1Root = parseYarnLockRawV1Root;
//# sourceMappingURL=v1.js.map