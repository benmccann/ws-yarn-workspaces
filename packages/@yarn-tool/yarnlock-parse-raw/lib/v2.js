"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stringifyYarnLockRawV2 = exports.parseYarnLockRawV2Root = exports.parseYarnLockRawV2 = void 0;
const parsers_1 = require("@yarnpkg/parsers");
Object.defineProperty(exports, "stringifyYarnLockRawV2", { enumerable: true, get: function () { return parsers_1.stringifySyml; } });
function parseYarnLockRawV2(source) {
    return (0, parsers_1.parseSyml)(source === null || source === void 0 ? void 0 : source.toString());
}
exports.parseYarnLockRawV2 = parseYarnLockRawV2;
exports.parseYarnLockRawV2Root = parseYarnLockRawV2;
//# sourceMappingURL=v2.js.map