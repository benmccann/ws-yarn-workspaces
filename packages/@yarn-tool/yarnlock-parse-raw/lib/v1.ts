import { parse, stringify } from '@yarnpkg/lockfile';
import { IYarnLockRawSourceV1, IYarnLockRawSourceV1Full } from '@yarn-tool/yarnlock-types';

declare module '@yarnpkg/lockfile'
{
	function parse<T extends IYarnLockRawSourceV1Full = IYarnLockRawSourceV1Full>(source: string, fileLoc?: string): T;
	function parse(source: string, fileLoc?: string): IYarnLockRawSourceV1Full;
}

export function parseYarnLockRawV1<T extends IYarnLockRawSourceV1Full = IYarnLockRawSourceV1Full>(source: string | Buffer, fileLoc?: string): T
{
	return parse(source?.toString(), fileLoc)
}

export function parseYarnLockRawV1Root(source: string | Buffer,
	fileLoc?: string
)
{
	return parseYarnLockRawV1(source, fileLoc).object
}

export { stringify as stringifyYarnLockRawV1 }
