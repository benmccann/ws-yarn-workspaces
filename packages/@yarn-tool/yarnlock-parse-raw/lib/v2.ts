import { parseSyml, stringifySyml } from '@yarnpkg/parsers';
import { IYarnLockRawSourceV2 } from '@yarn-tool/yarnlock-types';
import { EnumYarnLockBanner } from '@yarn-tool/yarnlock-banner';
import { EnumLineBreak } from 'crlf-normalize';

declare module '@yarnpkg/parsers'
{
	function parseSyml<T extends IYarnLockRawSourceV2 = IYarnLockRawSourceV2>(source: string): T
	function parseSyml(source: string): IYarnLockRawSourceV2
}

export function parseYarnLockRawV2<T extends IYarnLockRawSourceV2 = IYarnLockRawSourceV2>(source: string | Buffer): T
{
	return parseSyml(source?.toString()) as any
}

export { parseYarnLockRawV2 as parseYarnLockRawV2Root }

export function stringifyYarnLockRawV2(json: any, noHeader?: boolean)
{
	const list = [EnumYarnLockBanner.v2, stringifySyml(json)];

	if (noHeader)
	{
		list.shift();
	}

	return list.join(EnumLineBreak.LF + EnumLineBreak.LF)
}
