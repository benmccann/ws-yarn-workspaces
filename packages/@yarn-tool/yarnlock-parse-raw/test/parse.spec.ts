//@noUnusedParameters:false

import { join } from 'path';
import { __TEST_YARNLOCK } from '../../../../__root_ws';
import { EnumDetectYarnLock } from '@yarn-tool/yarnlock-types';
import { pathExistsSync } from 'fs-extra';
import { readFileSync } from 'fs';
import { checkResolutionsUpdate } from '@yarn-tool/ncu';
import { parseYarnLockRawV1Root, stringifyYarnLockRawV1 } from '../lib/v1';
import { parseYarnLockRawV2Root, stringifyYarnLockRawV2 } from '../lib/v2';
import { detectYarnLockVersionByObject } from '@yarn-tool/detect-yarnlock-version/lib/detectYarnLockVersionByObject';
import { _forEachVersionTags } from '../../../../test/lib/forEachVersionTags';

beforeAll(async () =>
{

});

describe(`parseYarnLockRaw`, () =>
{
	const dir = join(__TEST_YARNLOCK);

	_forEachVersionTags().forEach(ver =>
	{
		const file = join(dir, ver, 'yarn.lock');
		let fn: typeof parseYarnLockRawV1Root | typeof parseYarnLockRawV2Root
		let fn2: typeof stringifyYarnLockRawV1 | typeof stringifyYarnLockRawV2

		switch (ver)
		{
			case 'v2':
			case 'v3':
				fn = parseYarnLockRawV2Root;
				fn2 = stringifyYarnLockRawV2;
				break;
			default:
				fn = parseYarnLockRawV1Root;
				fn2 = stringifyYarnLockRawV1;
				break;
		}

		pathExistsSync(file) && test(ver, async () =>
		{
			const content = readFileSync(file).toString();

			let actual = fn(content);
			let actual2 = detectYarnLockVersionByObject(actual);

			const expected2 = EnumDetectYarnLock[ver];

			expect(actual2).toStrictEqual(expected2)
			expect(actual2).toMatchSnapshot();

			expect(actual).not.toHaveProperty('object');

			const content2 = fn2(actual);

			expect(content2).toStrictEqual(content)

		});
	});

})
