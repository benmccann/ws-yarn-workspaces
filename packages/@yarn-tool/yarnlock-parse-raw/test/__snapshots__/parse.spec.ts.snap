// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseYarnLockRaw v1 1`] = `1`;

exports[`parseYarnLockRaw v2 1`] = `2`;

exports[`parseYarnLockRaw v3 1`] = `3`;
